tests/acceptance/17_packages/10_new/unsafe/packagesmatching.cf

```diff
@@ -0,0 +1,55 @@
+# Test that updates list cache is not updated too often.
+
+body common control
+{
+    inputs => { "../../../default.cf.sub" };
+    bundlesequence => { default($(this.promise_filename)) };
+
+    package_inventory => { test_module };
+    #package_module => "test_module";
+}
+
+bundle agent init
+{
+  meta:
+    "test_skip_needs_work" string => "!linux";
+
+  files:
+    test_pass_1::
+      "$(sys.workdir)/modules/packages/test_module"
+        copy_from => local_cp("$(this.promise_dirname)/test_module"),
+        perms => m("0755");
+}
+
+bundle agent test
+{
+  commands:
+    test_pass_1::
+      "$(sys.cf_agent) -D AUTO -f $(this.promise_filename).sub";
+
+  vars:
+    test_pass_2::
+      "all_packages" data => packagesmatching(".*", ".*", ".*", ".*");
+      "all_packages_print" string => format("%S", "all_packages");
+      "test_module_packages" data => packagesmatching(".*", ".*", ".*", "test_module");
+      "test_module_packages_print" string => format("%S", "test_module_packages");
+}
+
+bundle agent check
+{
+  methods:
+    test_pass_1::
+      "any" usebundle => dcs_wait($(this.promise_filename), 0);
+    test_pass_2::
+      "all_ok" expression => strcmp("$(test.all_packages_print)", "[{"arch":"amd64","method":"test_module","name":"cfe-present-package","version":"3"}]");
+      "module_ok" expression => strcmp("$(test.test_module_packages_print)", "[{"arch":"amd64","method":"test_module","name":"cfe-present-package","version":"3"}]");
```

Comment:
Here it would be nice if you could do three steps. In each pass:

1. Do nothing, just let the package inventory update itself.
2. Check that version 1 is installed, and update to version 3.
3. Check that version 3 is installed.

This way we also test that it works without specifying a package promise, which has been a major complaint in the past.